Fixed in 4.5.2.

http://permalink.gmane.org/gmane.comp.gcc.devel/117591

	PR tree-optimization/46491
	Backport from mainline

	2010-05-14  Jan Hubicka  <jh@suse.cz>
	* ipa-pure-const.c (check_stmt): Do not use memory_identifier_string.


--- a/gcc/ipa-pure-const.c
+++ b/gcc/ipa-pure-const.c
@@ -453,7 +453,7 @@ check_stmt (gimple_stmt_iterator *gsip, funct_state local, bool ipa)
       for (i = 0; i < gimple_asm_nclobbers (stmt); i++)
 	{
 	  tree op = gimple_asm_clobber_op (stmt, i);
-	  if (simple_cst_equal(TREE_VALUE (op), memory_identifier_string) == 1)
+	  if (strcmp (TREE_STRING_POINTER (TREE_VALUE (op)), "memory") == 0)
 	    {
               if (dump_file)
                 fprintf (dump_file, "    memory asm clobber is not const/pure");
